content/post/111-js-ipfs-0-51-0.md

```diff
@@ -0,0 +1,220 @@
+---
+date: 2020-10-29
+url: /2020-10-29-js-ipfs-0-50/
+title: js-IPFS 0.51.0 adds type definitions and removes SECIO
+header_image: js-ipfs-placeholder.png
+author: Alex Potsides
+---
+
+# 🔦 Highlights
+
+> Type Defs! No more SECIO! Halve your install time!
+
+`js-IPFS@0.51.0` has taken off from the launchpad and is on route to your node_modules folder with TypeScript support, Custom DAG Formats and a new, lighter way to consume the module
+
+## 🍪 Types
+
+TypeScript has taken the JS world by storm, allowing developers to be alerted when the interface they are coding against is not what they expect and to enable vast amounts of useful tooling like accurate code completion and inline doc viewing.
+
+There have been [several](https://github.com/vishalkuo/types-ipfs) [community](https://github.com/survirtual/types-ipfs) [led](https://github.com/beenotung/typestub-ipfs) [efforts](https://github.com/zabirauf/ipfs-types) to provide type information for `js-IPFS` but since the project moves so quickly it's an uphill struggle to keep compatibility with the changes that get landed in every release.
+
+With `js-IPFS@0.51.0` types are now a first-class citizen within the codebase.  The project has not gone so far as to convert to TypeScript - we do not wish knowledge of TypeScript to become a prerequisite to contributing to `js-IPFS` development, instead all types are defined using [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) comments and are built and validated during development and Continuous Integration.
+
+<p style="max-width:1000px;margin-left:auto;margin-right:auto;">
+<img src="/111-js-ipfs-0-51-0/types.png">
+</p>
+
+## SECIO Removal
```

Comment:
Is there any impact on devs using js-ipfs? Any actions needed? If so, should summarize here.

Reply:
Good shout, I've added a paragraph about upgrading.